subsys/logging/log_core.c

```diff
@@ -1215,6 +1223,31 @@ bool z_log_msg2_pending(void)
 	return mpsc_pbuf_is_pending(&log_buffer);
 }
 
+const char *z_log_get_tag(void)
+{
+	return CONFIG_LOG_TAG_MAX_LEN > 0 ? tag : NULL;
+}
+
+int log_set_tag(const char *str)
+{
+	if (CONFIG_LOG_TAG_MAX_LEN == 0) {
+		return -ENOTSUP;
+	}
+
```

Comment:
I recommend checking if  `str` is not NULL because `strlen` does not do it and it will return a random value.